components/table/Column.razor.cs

```diff
@@ -136,9 +152,39 @@ protected override void OnInitialized()
             Sortable = Sortable || SortModel != null;
             _sortDirection = SortModel?.SortDirection ?? DefaultSortOrder ?? SortDirection.None;
 
+            _columnDataType = THelper.GetUnderlyingType<TData>();
+            if (_columnDataType == typeof(bool))
+            {
+                if (Filters == null || Filters?.Any() != true)
+                {
+                    Filters = new List<TableFilter<TData>>();
+
+                    var yesFilter = GetNewFilter();
+                    yesFilter.Text = "True";
+                    yesFilter.Value = (TData)Convert.ChangeType(true, typeof(bool));
```

Comment:
Why would you call Convert.ChangeType to convert the same type?

Reply:
i need to convert bool to TData ,if there is another way to do it then please help.